confz/loaders/loader.py

```diff
@@ -42,7 +42,7 @@ def transform_nested_dicts(
         """
         dict_out: Dict[str, Any] = {}
         for key, value in dict_in.items():
-            if separator in key:
+            if separator in key and not key.startswith(separator):
```

Comment:
What was the reason for this additional check?

Reply:
In my system I have some envs that start with `__` (something related to OS X) and that resulted in creating bad mappings in `dict_out` as `separator in key` was `True` but the env itself was not valid nested variable. So I decided to add this additional check in order to avoid error this error.

Reply:
Ok, I see